README.md

```diff
@@ -79,13 +107,24 @@ API reference
 API overview with typeScript type annotations is found below. Associated browser Web IDL can be found [here](https://source.chromium.org/chromium/chromium/src/+/master:third_party/blink/renderer/modules/url_pattern/).
 
 ```ts
+type URLPatternInput = URLPatternInit | string;
+
 class URLPattern {
-  constructor(init: URLPatternInit);
-  constructor(shortPattern: string, baseURL: string = ""));
+  constructor(init?: URLPatternInput, baseURL?: string);
 
-  test(input: URLPattern | string): boolean;
-  exec(input: URLPattern | string): URLPatternResult;
-};
+  test(input?: URLPatternInput, baseURL?: string): boolean;
+
+  exec(input?: URLPatternInput, baseURL?: string): URLPatternResult | null;
+
+  get protocol(): string;
```

Comment:
nit: might be _slightly_ nicer to write these as readonly properties:
```ts
readonly protocol: string;
```

Reply:
Done